frontend/src/metabase/query_builder/components/notebook/steps/DataStep/DataStep.unit.spec.tsx

```diff
@@ -0,0 +1,139 @@
+import userEvent from "@testing-library/user-event";
+import { getIcon, renderWithProviders, screen } from "__support__/ui";
+import { createSampleDatabase } from "metabase-types/api/mocks/presets";
+import {
+  setupDatabasesEndpoints,
+  setupSearchEndpoints,
+} from "__support__/server-mocks";
+import * as Lib from "metabase-lib";
+import { columnFinder, createQuery } from "metabase-lib/test-helpers";
+import { createMockNotebookStep } from "../../test-utils";
+import { DataStep } from "./DataStep";
+
+const createQueryWithFields = (columnNames: string[]) => {
+  const query = createQuery();
+  const findColumn = columnFinder(query, Lib.fieldableColumns(query, 0));
+  const columns = columnNames.map(name => findColumn("ORDERS", name));
+  return Lib.withFields(query, 0, columns);
+};
+
+const setup = async (step = createMockNotebookStep()) => {
+  const updateQuery = jest.fn();
+  setupDatabasesEndpoints([createSampleDatabase()]);
+  setupSearchEndpoints([]);
+
+  renderWithProviders(
+    <DataStep
+      step={step}
+      topLevelQuery={step.topLevelQuery}
+      query={step.query}
+      color="brand"
+      isLastOpened={false}
+      reportTimezone="UTC"
+      updateQuery={updateQuery}
+    />,
+  );
+
+  const getNextQuery = (): Lib.Query => {
```

Comment:
Follows the conventions established by other steps, e.g. `BreakoutStep.unit.spec.tsx`